records_mover/records/types.py

```diff
@@ -86,18 +75,100 @@ class UrlDetailsEntry(TypedDict):
 
     DelimitedVariant = str
 
+if TYPE_CHECKING:
+    HintEncoding = Literal["UTF8", "UTF16", "UTF16LE", "UTF16BE",
+                           "UTF16BOM", "UTF8BOM", "LATIN1", "CP1252"]
+else:
     HintEncoding = str
 
+VALID_ENCODING: List[HintEncoding] = [
+    "UTF8", "UTF16", "UTF16LE", "UTF16BE", "UTF16BOM", "UTF8BOM", "LATIN1", "CP1252"
+]
+
+if TYPE_CHECKING:
+    HintQuoting = Literal["all", "minimal", "nonnumeric", None]
+else:
     HintQuoting = Optional[str]
 
+VALID_QUOTING: List[HintQuoting] = ["all", "minimal", "nonnumeric", None]
+
+
+if TYPE_CHECKING:
+    HintEscape = Literal["\\", None]
+else:
     HintEscape = Optional[str]
 
+VALID_ESCAPE: List[HintEscape] = ["\\", None]
+
+# TODO: combine this and cli thingie
+if TYPE_CHECKING:
+    HintCompression = Literal['GZIP', 'BZIP', 'LZO', None]
+else:
     HintCompression = Optional[str]
 
+VALID_COMPRESSIONS: List[HintCompression] = ['GZIP', 'BZIP', 'LZO', None]
+
+if TYPE_CHECKING:
+    # The trick here works on Literal[True, False] but not on bool:
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker